app/models/book.rb

```diff
@@ -80,150 +80,147 @@ def self.bulk_upsert(rows)
     # Duplicate object IDs will be refused due to the unique index on obj_id.
     rows.uniq!{ |r| r[:obj_id] }
     return if rows.empty?
-
     sql = StringIO.new
-
+    
     sql << 'INSERT INTO books('
     sql << COLUMNS.join(', ')
     sql << ') VALUES '
-
+    
     value_index = 0
     rows.length.times do |index|
       sql << "\n\t("
       sql << COLUMNS.map{ |c| "$#{value_index += 1}" }.join(', ')
       sql << ')'
       sql << ',' if index < rows.length - 1
     end
-
+    
     not_null_bool_cols = %w(exists_in_google
-        exists_in_hathitrust exists_in_internet_archive)
-
-    binds = []
```

Comment:
I think there are an extra 2 spaces of indentation starting at this point

Reply:
The indentation should now be correct